packages/web-components/src/badge/badge.styles.ts

```diff
@@ -1,259 +1,293 @@
-        }
-        :host([appearance="tint"][color="subtle"]) {
-            background-color: ${colorNeutralBackground1};
-            color: ${colorNeutralForeground3};
-            border-color: ${colorNeutralStroke2};
-        }
-        :host([appearance="tint"][color="success"]) {
-            background-color: ${colorPaletteGreenBackground1};
-            color: ${colorPaletteGreenForeground1};
-            border-color: ${colorPaletteGreenBackground2};
-        }
-        :host([appearance="tint"][color="warning"]) {
-            background-color: ${colorPaletteYellowBackground1};
-            color: ${colorPaletteYellowForeground2};
-            border-color: ${colorPaletteYellowBackground2};
-        }
-    `),
-    sizeBehavior("tiny", css`
-        :host([size="tiny"]) {
-            width: 6px;
-            height: 6px;
-            font-size: 4px;
-        }
-        ::slotted(svg) {
-            font-size: 6px;
-        }
-    `),
-    sizeBehavior("extra-small", css`
-        :host([size="extra-small"]) {
-            width: 10px;
-            height: 10px;
-            font-size: 6px;
-        }
-        ::slotted(svg) {
-            font-size: 10px;
-        }
-    `),
-    sizeBehavior("small", css`
-        :host([size="small"]) {
-            min-width: 16px;
-            height: 16px;
-            font-size: 8px;
-            padding: 2px;
-            gap: 4px;
-        }
-        ::slotted(svg) {
-            font-size: 12px;
-        }
-    `),
-    sizeBehavior("medium", css`
-        :host([size="medium"]) {
-            min-width: 20px;
-            height: 20px;
-            font-size: 10px;
-            padding: 4px;
-            gap: 4px;
-        }
-        ::slotted(svg) {
-            font-size: 12px;
-        }
-    `),
-    sizeBehavior("large", css`
-        :host([size="large"]) {
-            min-width: 24px;
-            height: 24px;
-            font-size: 12px;
-            padding: 4px;
-            gap: 4px;
-        }
-        ::slotted(svg) {
-            font-size: 16px;
-        }
-    `),
-    sizeBehavior("extra-large", css`
-        :host([size="extra-large"]) {
-            min-width: 32px;
-            height: 32px;
-            font-size: 12px;
-            padding: 6px;
-            gap: 6px;
-            border-width: ${strokeWidthThick};
-        }
-        ::slotted(svg) {
-            font-size: 20px;
-        }
-    `),
-    shapeBehavior("rounded", css`
-        :host([shape="rounded"]) {
-            border-radius: ${borderRadiusMedium};
-        }
-        :host([shape="rounded"][size="tiny"]),
-        :host([shape="rounded"][size="extra-small"]),
-        :host([shape="rounded"][size="small"]) {
-            border-radius: ${borderRadiusSmall};
-        }
-    `),
-    shapeBehavior("circular", css`
-        :host([shape="circular"]) {
-            border-radius: ${borderRadiusCircular};
-        }
-    `)
-);
+  ${display('inline-flex')} :host {
```

Comment:
This change is mostly whitespace tabbing the styles to the left. The notable change is for learning.

I've removed the `withBehaviors` call (and the utils/* directory containing behaviors). Looking through code and comments it seems the behaviors add/remove stylesheets conditionally. Including all the styles up front seemed to be an acceptably naive and simple solution.

Would like to learn what problems the behaviors were created to solve.

Reply:
See thread https://github.com/chrisdholt/fluentui/pull/3#discussion_r971020535 in this PR.